hs/src/Reach/Eval/Core.hs

```diff
@@ -2771,13 +2772,13 @@ evalPrim p sargs =
           [x] -> return (x, Nothing)
           [x, y] -> return (x, Just y)
           _ -> illegal_args
-      toka <- compileCheckType T_Token tokv
       at <- withAt id
+      toka <- compileCheckType (Just at) T_Token tokv
```

Comment:
This is actually a "by definition", unless you want to use the srcloc of the call as a the source for "by definition"s in general

Reply:
Yeah, I used the call as the source for things that were “by definition” generally.  I didn't hit all of them just because I wasn't immediately sure what each type check was for, and wanted to get to a PR to get feedback before spending more time figuring them out.  I think I'll switch the ones where I used the call site to have a “definition” source location.